### PR TITLE
[fix][broker] PulsarResourceDescription.compareTo should return zero on exactly match

### LINKED ISSUE #11579: Fix the compareTo method for PulsarResourceDescription
**Describe the bug**

**To Reproduce**

**Expected behavior**

need to return zero if two resourceDescription match exactly

**Screenshots**

**Desktop (please complete the following information):**

**Additional context**
I will fix for this。

### PR DESCRIPTION
**Pull request by leizhiyuan**
Fixes #11579

### Motivation

need to return zero if two resourceDescription match exactly

### Modifications

add a method named exactlyEquals to fix this issue

### Documentation

- [x] `doc-not-needed`

### CODE REVIEW

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/loadbalance/impl/PulsarResourceDescriptionTest.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package org.apache.pulsar.broker.loadbalance.impl;
```
Comment (kezhenxu94) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/loadbalance/impl/PulsarResourceDescriptionTest.java:1]: license header is missing

Comment (leizhiyuan) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/loadbalance/impl/PulsarResourceDescriptionTest.java:19]: done

**Comment by Anonymitaet** [comment]
Thanks for providing doc-related info!

**Comment by codelipenghui** [comment]
The pr had no activity for 30 days, mark with Stale label.

**Review by tisonkun — APPROVED** [review]
LGTM

**Comment by tisonkun** [comment]
Merging...

Thanks for your contribution @leizhiyuan!

**Pull request merged**